### PR TITLE
Add support for dump all table entries with table get command

### LINKED ISSUE #23: Implement dumping whole table for 'table get' command
For now `table get` is able to only dump one entry with known key.

### PR DESCRIPTION
**Pull request by tatry**
Closes #23 

Allows to dump all entries from a table. Default entry will be added in a separate PR.

### CODE REVIEW

File: CLI/table.c [hunk 1]
```diff
@@ -874,19 +889,23 @@ int do_table_get(int argc, char **argv)
...
-    if (error_code != NO_ERROR)
-        goto clean_up;
-
-    error_code = print_json_table_entry(&ctx, &entry, table_name);
+    if (key_provided) {
+        error_code = psabpf_table_entry_get(&ctx, &entry);
+        if (error_code != NO_ERROR)
+            goto clean_up;
+    }
+    error_code = print_json_table_entry(&ctx, key_provided ? &entry : NULL, table_name);
```
Comment (kmateuszssak) [inline CLI/table.c:908]: Using `else` avoids second check `key_provided ? &entry`

File: lib/psabpf_table.c [hunk 2]
```diff
@@ -2568,14 +2581,312 @@ int psabpf_table_entry_get(psabpf_table_entry_ctx_t *ctx, psabpf_table_entry_t *
...
+
+    if (field_type == PSABPF_TERNARY) {
+        psabpf_matchkey_type(&mk, PSABPF_TERNARY);
+        psabpf_matchkey_data(&mk, field_data, field_len);
+        psabpf_matchkey_mask(&mk, field_mask, field_len);
+    } else if (field_type == PSABPF_LPM) {
+        psabpf_matchkey_type(&mk, PSABPF_LPM);
+        psabpf_matchkey_data(&mk, field_data, field_len);
+        /* Convert network byte order into host order */
+        for (size_t i = 0; i < mk.key_size / 2; ++i) {
```
Comment (kmateuszssak) [inline lib/psabpf_table.c:2647]: Could we move it as new utility function?

**Review by kmateuszssak — COMMENTED** [review]
Few questions.

File: CLI/table.c (hunk 1 above)
Comment (tatry) [reply CLI/table.c:908]: With `else` funtion `print_json_table_entry` must be called twice, one in `if` and one in `else` block. Now I think that better name for this function will be `print_json_table`. `entry` is provided only when single entry must be printed.

File: lib/psabpf_table.c [hunk 3]
```diff
@@ -2568,14 +2581,312 @@ int psabpf_table_entry_get(psabpf_table_entry_ctx_t *ctx, psabpf_table_entry_t *
...
+    /* Iterate in order assumed by data plane algorithm */
+    if (ctx->current_raw_key_mask == NULL)
+        ctx->current_raw_key_mask = calloc(1, ctx->prefixes.key_size);
+
+    if (prefix_value == NULL || next_key == NULL || ctx->current_raw_key_mask == NULL) {
+        ret_code = ENOMEM;
+        goto clean_up;
+    }
+
+    while (true) {
```
[the remark this answers is not in the document]
Comment (tatry) [reply lib/psabpf_table.c:2749]: It is doable, but condition will look awful in my opinion (not tested yet):
```c
while (ctx->table.fd < 0 || (ctx->table.fd >= 0 && bpf_map_get_next_key(ctx->table.fd, ctx->current_raw_key, next_key) != 0))
```

So, I prefer the `while (true)` in this case.

**Review by kmateuszssak — APPROVED** [review]

File: lib/common.c [hunk 4]
```diff
@@ -42,6 +42,17 @@ void mem_bitwise_and(uint32_t *dst, uint32_t *mask, size_t len)
     }
 }
 
+void swap_byte_order(uint8_t *data, size_t len)
```
Comment (osinstom) [inline lib/common.c:45]: I belive that `char *` is commonly preferred to pass a pointer to the beginning of data, instead of `uint8_t *`

**Review by osinstom — APPROVED** [review]
Left a minor comment, but generally LGTM

File: lib/common.c (hunk 4 above)
Comment (tatry) [reply lib/common.c]: Done

**Pull request merged**